tools/scion/scion-topo-sanitizer.py

```diff
@@ -0,0 +1,96 @@
+                        brd = routers[router_name]
+                        internal_addr = brd['internal_addr']
+                        interfaces = brd['interfaces']
+                        for i, (id,intf) in enumerate(interfaces.items()):
+                            underlay = intf['underlay']
+                            remote = intf['isd_as']
+                            remote_asn = remote.split('-')[1]
+                            assert remote != asn
+                            local_ip = underlay['local'].split(':')[0] #.rstrip(':50000')
+                            remote_ip = underlay['remote'].split(':')[0]
+
+                            brd_service = services[folder]
+
+                            router_nets = brd_service['networks']
+                            for  (name,net) in router_nets.items():
+
+                                if name.endswith('net0'): # local network within AS
+                                    ip = net['ipv4_address']
+                                    #net_name = name.lstrip('net_').rstrip('_net0')
+                                    net_name = name.split('_')[1]
+                                    assert net_name == asn, f'border router must belong to a single AS only: {net_name} {asn} [{name}]'
+                                    router_internal_ip = internal_addr.split(':')[0] #.rstrip(':30042')
+                                    assert ip==router_internal_ip, f'contradiction between topology.json and docker-compose.yml detected: {name} AS {asn} expected: {router_internal_ip} actual: {ip}'
+
+                                if name.startswith('net_ix'):
+                                    ip = net['ipv4_address']
+                                    net_name = name.split('_')[-1].lstrip('ix')
+                                    assert ip == local_ip, f'contradiction between topology.json and docker-compose.yml detected: {name}'
+                                    # TODO: combine net_name and remote to get remote-border-router 
+                                    # then check that remote-border-router has 'name' under 'networks'
+                                    # and within this networks in fact the IP 'remote_ip'
+                                    remote_router_name = f'brdnode_{remote_asn}_router{net_name}'
+                                    remote_router = services[remote_router_name]
+                                    remote_net = remote_router['networks'][name]
+                                    assert remote_net['ipv4_address'] == remote_ip  , f'contradiction between topology.json and docker-compose.yml detected: {name} AS {remote_asn} router{net_name}'
+                                    pass
+                            pass
+                        pass 
+
+                except Exception as e:
+                   print(f"Error reading {file_path}: {e}")
+    print('No errors detected')
+
+if __name__ == "__main__":
+    find_and_check_json_files()
```

Comment:
The directory that is searched for topology files could easily be passed on the command line.